.github/workflows/test_periodic.yml

```diff
@@ -0,0 +1,226 @@
+
+    - name: Cache Kubernetes
+      id: cache-k8s
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.GOPATH }}/src/k8s.io/kubernetes/
+        key: kubernetes-${{ env.K8S_VERSION }}
+
+    - name: Set up Go
+      uses: actions/setup-go@v1
+      with:
+        go-version: ${{ env.GO_VERSION }}
+      if: steps.cache-k8s.outputs.cache-hit != 'true'
+
+    - name: Build and install Kubernetes
+      if: steps.cache-k8s.outputs.cache-hit != 'true'
+      run: |
+        set -x
+        rm -rf $GOPATH/src/k8s.io/kubernetes
+        git clone --single-branch --branch $K8S_VERSION https://github.com/kubernetes/kubernetes.git $GOPATH/src/k8s.io/kubernetes/
+        pushd $GOPATH/src/k8s.io/kubernetes/
+          make WHAT="test/e2e/e2e.test vendor/github.com/onsi/ginkgo/ginkgo cmd/kubectl"
+          rm -rf .git
+        popd
+
+  e2e-dual:
+    name: e2e-dual
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - shard: shard-n-other
+          - shard: shard-np
+          - shard: control-plane
+        ha:
+         - enabled: "true"
+           name: "HA"
+         - enabled: "false"
+           name: "noHA"
+    needs: [build, k8s]
+    env:
+      JOB_NAME: "${{ matrix.target.shard }}-${{ matrix.ha.name }}"
+      KIND_HA: "${{ matrix.ha.enabled }}"
+      # Deploy dual stack cluster
+      KIND_IPV4_SUPPORT: true
+      KIND_IPV6_SUPPORT: true
+    steps:
+
+    - name: Set up Go
+      uses: actions/setup-go@v1
+      with:
+        go-version: ${{ env.GO_VERSION }}
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Set up environment
+      run: |
+        export GOPATH=$(go env GOPATH)
+        echo "::set-env name=GOPATH::$GOPATH"
+        export PATH=$GOPATH/bin:$PATH
+        echo "::add-path::$GOPATH/bin"
+
+    - name: Restore cache Kubernetes
+      id: cache-k8s
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.GOPATH }}/src/k8s.io/kubernetes/
+        key: kubernetes-${{ env.K8S_VERSION }}
+
+    # Re-build if kube wasn't in the cache due to
+    # https://github.com/actions/cache/issues/107#issuecomment-598188250
+    # https://github.com/actions/cache/issues/208#issuecomment-596664572
+    - name: Build and install Kubernetes
+      if: steps.cache-k8s.outputs.cache-hit != 'true'
+      run: |
+        set -x
+        rm -rf $GOPATH/src/k8s.io/kubernetes
+        git clone --single-branch --branch $K8S_VERSION https://github.com/kubernetes/kubernetes.git $GOPATH/src/k8s.io/kubernetes/
+        pushd $GOPATH/src/k8s.io/kubernetes/
+          make WHAT="test/e2e/e2e.test vendor/github.com/onsi/ginkgo/ginkgo cmd/kubectl"
+          rm -rf .git
+        popd
+
+    - uses: actions/download-artifact@v2
+      with:
+        name: test-image
+    - name: Load docker image
+      run: |
+        docker load --input image.tar
+
+    - name: kind setup
```

Comment:
Will this be replaced with `install-kind.sh` at some time in the future?  Assume you need additional patches to KIND for dual-stack (https://github.com/aojea/kind/releases/download/dualstack/kind).

Reply:
yeah, but right now is a chicken an egg problem, we need to stabilize dual-stack first.
This way we avoid any risk of affecting other things in the repo, and allows us to have signal on master